File: Jim-and-the-Orders.ts
'use strict';

import { WriteStream, createWriteStream } from "fs";
process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString: string = '';
let inputLines: string[] = [];
let currentLine: number = 0;

process.stdin.on('data', function(inputStdin: string): void {
    inputString += inputStdin;
});

process.stdin.on('end', function(): void {
    inputLines = inputString.split('\n');
    inputString = '';

    main();
});

function readLine(): string {
    return inputLines[currentLine++];
}

/*
 * Complete the 'jimOrders' function below.
 *
 * The function is expected to return an INTEGER_ARRAY.
 * The function accepts 2D_INTEGER_ARRAY orders as parameter.
 */

function jimOrders(orders: number[][]): number[] {
    return orders
        .map((order, index) => [order[0] + order[1], index + 1])
        .sort((a, b) => a[0] - b[0] || a[1] - b[1])
        .map(item => item[1]);
}


function jimOrders2(orders: number[][]): number[] {
    const deliveryTimes = orders.map((order, index) => {
        const totalTime = order[0] + order[1];
        const orderId = index + 1;
        return [totalTime, orderId];
    });

    deliveryTimes.sort((a, b) => {
        if (a[0] !== b[0]) {
            return a[0] - b[0];
        }
        return a[1] - b[1];
    });

    return deliveryTimes.map(item => item[1]);
}


function main() {
    const ws: WriteStream = createWriteStream(process.env['OUTPUT_PATH']);

    const n: number = parseInt(readLine().trim(), 10);

    let orders: number[][] = Array(n);

    for (let i: number = 0; i < n; i++) {
        orders[i] = readLine().replace(/\s+$/g, '').split(' ').map(ordersTemp => parseInt(ordersTemp, 10));
    }

    const result: number[] = jimOrders(orders);

    ws.write(result.join(' ') + '\n');

    ws.end();
}

File: Jim-and-the-Orders.go
package main

import (
    "bufio"
    "fmt"
    "io"
    "os"
    "strconv"
    "strings"
    "sort"
)

/*
 * Complete the 'jimOrders' function below.
 *
 * The function is expected to return an INTEGER_ARRAY.
 * The function accepts 2D_INTEGER_ARRAY orders as parameter.
 */

func jimOrders(orders [][]int32) []int32 {
   type orderInfo struct {
        totalTime int32
        orderID   int32
    }    

    arr := make([]orderInfo, len(orders))

    for i, order := range orders {
        arr[i] = orderInfo{
            totalTime: order[0] + order[1],
            orderID:   int32(i + 1),
        }
    }    

    sort.Slice(arr, func(i, j int) bool {
        if arr[i].totalTime != arr[j].totalTime {
            return arr[i].totalTime < arr[j].totalTime
        }
        return arr[i].orderID < arr[j].orderID
    })
    
}


func main() {
    reader := bufio.NewReaderSize(os.Stdin, 16 * 1024 * 1024)

    stdout, err := os.Create(os.Getenv("OUTPUT_PATH"))
    checkError(err)

    defer stdout.Close()

    writer := bufio.NewWriterSize(stdout, 16 * 1024 * 1024)

    nTemp, err := strconv.ParseInt(strings.TrimSpace(readLine(reader)), 10, 64)
    checkError(err)
    n := int32(nTemp)

    var orders [][]int32
    for i := 0; i < int(n); i++ {
        ordersRowTemp := strings.Split(strings.TrimRight(readLine(reader)," \t\r\n"), " ")

        var ordersRow []int32
        for _, ordersRowItem := range ordersRowTemp {
            ordersItemTemp, err := strconv.ParseInt(ordersRowItem, 10, 64)
            checkError(err)
            ordersItem := int32(ordersItemTemp)
            ordersRow = append(ordersRow, ordersItem)
        }

        if len(ordersRow) != 2 {
            panic("Bad input")
        }

        orders = append(orders, ordersRow)
    }

    result := jimOrders(orders)

    for i, resultItem := range result {
        fmt.Fprintf(writer, "%d", resultItem)

        if i != len(result) - 1 {
            fmt.Fprintf(writer, " ")
        }
    }

    fmt.Fprintf(writer, "\n")

    writer.Flush()
}
